### PR TITLE
Elasticsearch

### PR DESCRIPTION
**Pull request by tarek-nawara**
Finished Classifying tweets using the existing model and added integration with Elasticsearch for future visualization.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,13 @@
 
 # sentiment-analysis
 Simple show case of consuming twitter api and performing sentiment analysis over tweets
+
+### How to contribute
+You need the following dependencies installed:
+- [Gradle](https://gradle.org/)
+- [ElasticSearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/_installation.html)
```
Comment (mohamedramadan2515) [inline README.md]: can we add this for Windows users (https://www.elastic.co/downloads/elasticsearch)

File: README.md [hunk 2]
```diff
@@ -2,3 +2,13 @@
...
+- [Kibana](https://www.elastic.co/downloads/kibana)
+
+### How to run it
+After installing `Elasticsearch` just go to the bin directory and fire up `elasticsearch`
+after that you need to run `gradle run`
```
Comment (mohamedramadan2515) [inline README.md]: also for windows users run bin/elasticsearch

File: src/main/scala/edu/twitter/classification/Classifier.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      tweet <- tweets
+      if supportedLangIso(tweet.getLang)
+      features = hashingTF.transform(tweet.getText.split(" "))
+      label = model(features)
+    } yield ClassifiedTweet(label, tweet.getText)
+
+    classifiedStream
+  }
+
+  /** Create the Classification model. */
```
Comment (mohamedramadan2515) [inline src/main/scala/edu/twitter/classification/Classifier.scala]: add @return

File: src/main/scala/edu/twitter/classification/Classifier.scala [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    * and return a stream of `ClassifiedTweets`.
+    *
+    * @return stream of `ClassifiedTweets`
+    */
+  def createClassifiedStream(): DStream[ClassifiedTweet] = {
+    val tweets = new TwitterStream(ssc).createStream()
+    val model = createModel()
+    val hashingTF = new HashingTF(2000)
+    val supportedLangIso = Set("en", "eng")
+    val classifiedStream = for {
```
Comment (mohamedramadan2515) [inline src/main/scala/edu/twitter/classification/Classifier.scala]: Is this equal to reduce or collect in the stream api?

File: src/main/scala/edu/twitter/model/GradientBoostingModel.scala [hunk 5]
```diff
@@ -35,7 +37,7 @@ class GradientBoostingModel(trainingData: RDD[LabeledPoint], validationData: RDD
 
     var model = GradientBoostedTrees.train(trainingData, boostingStrategy)
     evaluate(model)
-    model
+    (features: org.apache.spark.mllib.linalg.Vector) => model.predict(features)
```
Comment (mohamedramadan2515) [inline src/main/scala/edu/twitter/model/GradientBoostingModel.scala]: nice trick 👍

File: README.md (hunk 1 above)
Comment (tarek-nawara) [reply README.md]: Sure, will do it.

File: README.md (hunk 2 above)
Comment (tarek-nawara) [reply README.md]: Sure, will do it.

File: src/main/scala/edu/twitter/classification/Classifier.scala (hunk 4 above)
Comment (tarek-nawara) [reply src/main/scala/edu/twitter/classification/Classifier.scala]: No, for comprehensions in Scala is desugared to `flatmap` and `map`, these are all transformations to the original data not actions like the `collect`. the difference is that actions force the computation to happen, unlike transformations.

File: src/main/scala/edu/twitter/model/GradientBoostingModel.scala (hunk 5 above)
Comment (tarek-nawara) [reply src/main/scala/edu/twitter/model/GradientBoostingModel.scala]: Thanks :)

File: src/main/scala/edu/twitter/classification/Classifier.scala (hunk 3 above)
Comment (tarek-nawara) [reply src/main/scala/edu/twitter/classification/Classifier.scala]: Sure, will do it.

**Pull request merged**